### DISCUSSION TITLE
C Api - New function to determine command/query type

### DISCUSSION THREAD

**Discussion by Giorgi** [duckdb/duckdb#8786] [Ideas]
Given a string that contains one or more statements, when I extract statements and execute them with `duckdb_execute_prepared` I get `duckdb_result` but I want to return results only for the statements that were a `Select` statement. This will also help with the [Pivot issue](https://github.com/duckdb/duckdb/issues/7720)

A new function that tells the type of the command associated with `duckdb_result` would enable this feature.

**Comment by Mause** [comment]
There's two ways we could about this I guess

We could return the "result type", so NOTHING/CHANGED_ROWS/QUERY_RESULT

Edit: the Java version is here: https://github.com/duckdb/duckdb/blob/c21a9cb87cd7724d9556a6fa76cae3e1c6ee22e3/tools/jdbc/src/main/java/org/duckdb/StatementReturnType.java#L4

Or the statement type, INSERT/SELECT/UPDATE/etc

I think the first one is more useful?

**Reply by Tishj** [reply]
The latter would be more composable though
i.e from `statement_type` you can figure out `result_type`, but reverse isn't possible

**Reply by Mause** [reply]
Does that hold true for `INSERT ... RETURNING ..` statements?

**Reply by Giorgi** [reply]
If I have an Insert statement with a `RETURNING` clause what will be returned by these Enums? I need to return result set for such statements

**Reply by Tishj** [reply]
Ah right, yea I think the result_type enum would be better then because just the statement_type alone isnt enough to infer the result_type

(The result_type should cover this ^)

**Reply by Giorgi** [reply]
Actually it would be useful to have both :)

**Reply by Mause** [reply]
Can do!

**Reply by Giorgi** [reply]
Are there any plans to add these functions to the C api?

**Reply by Mause** [reply]
I have a branch with them and some other requested functions, need to get around to writing some more tests using them

**Comment by Giorgi** [comment]
@Mause I think this is completed right?

**Reply by Mause** [reply]
Yep!
